### PR TITLE
refactor: theming

### PR DESCRIPTION
**Pull request by MEsteves22**
- `HvProvider` refactored
- `createTheme` utility added 
- `useTheme` hook added 
- Get started and provider documentation reviewed 

Note: The theming documentation will be reviewed on another PR.

### CODE REVIEW

File: packages/core/src/components/BaseDropdown/BaseDropdown.tsx [hunk 1]
```diff
@@ -143,7 +142,7 @@ export const HvBaseDropdown = ({
   onContainerCreation,
   ...others
 }: HvBaseDropdownProps) => {
-  const { rootId } = useContext(HvThemeContext);
+  const { rootId } = useTheme();
```
Comment (nantunes) [inline packages/core/src/components/BaseDropdown/BaseDropdown.tsx:145]: This one seems strange coming from a `useTheme` hook (it was already, coming from a `HvThemeContext`). I understand why, and that it is purely theme/styling related, but for consumers it is more something like the app container.

A `useRoot` returning the `id` and possibly a `ref` for the dom element would be interesting.

Comment (nantunes) [reply packages/core/src/components/BaseDropdown/BaseDropdown.tsx:145]: Also, being in separate contexts would also allow changing the theme without triggering re-renders in components using the root element's id/reference, as it most likely never change.

File: packages/core/src/hooks/useTheme.ts
Comment (nantunes) [inline packages/core/src/hooks/useTheme.ts:1]: If the hook is exposing all the properties in the context, `return useContext(HvThemeContext);` is enough... The `useMemo` is overkill.

**Review by nantunes — APPROVED** [review]

File: packages/core/src/hooks/useTheme.ts
Comment (MEsteves22) [reply packages/core/src/hooks/useTheme.ts:1]: I changed the hook to not use the `useMemo` hook as you mentioned.

**Review by plagoa — APPROVED** [review]

**Pull request merged**